itest/lnd_channel_funding_utxo_selection_test.go

```diff
@@ -0,0 +1,309 @@
+	// channel.
+	feeRate btcutil.Amount
+
+	// expectedBalance is Alice's expected balance in her channel.
+	expectedBalance btcutil.Amount
+
+	// remainingWalletBalance is Alice's expected remaining wallet balance
+	// after she opened a channgel.
+	remainingWalletBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+
+	// expectedErrStr contains the expected error in case chanOpenShouldFail
+	// is set to true.
+	expectedErrStr string
+
+	// commitmentType allows to define the exact type when opening the
+	// channel.
+	commitmentType lnrpc.CommitmentType
+
+	// private denotes if the channel opening is announced to the network or
+	// not.
+	private bool
+
+	//
+	reuseUtxo bool
+}
+
+// testChannelUtxoSelection checks various channel funding scenarios where the
+// user instructed the wallet to use a selection funds available in the wallet.
+func testChannelUtxoSelection(ht *lntest.HarnessTest) {
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	args := lntest.NodeArgsForCommitType(lnrpc.CommitmentType_ANCHORS)
+	alice := ht.NewNode("Alice", args)
+	defer ht.Shutdown(alice)
+
+	bob := ht.NewNode("Bob", args)
+	defer ht.Shutdown(bob)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	ht.EnsureConnected(alice, bob)
+
+	// Calculate reserve amount for one channel.
+	reserveResp, _ := alice.RPC.WalletKit.RequiredReserve(
+		context.Background(), &walletrpc.RequiredReserveRequest{
+			AdditionalPublicChannels: 1,
+		},
+	)
+
+	reserveAmount := btcutil.Amount(reserveResp.RequiredReserve)
+
+	var tcs = []*chanFundUtxoSelectionTestCase{
+		// Selected coins would leave a dust output after subtracting
+		// miner fees.
+		{
+			name:               "fundmax, wallet amount is dust",
+			initialCoins:       []btcutil.Amount{2_000},
+			selectedCoins:      []btcutil.Amount{2_000},
+			chanOpenShouldFail: true,
+			feeRate:            15,
+			expectedErrStr: "output amount(0.00000174 BTC) after " +
+				"subtracting fees(0.00001826 BTC) below dust " +
+				"limit(0.0000033 BTC)",
+		},
+		// Selected coins don't cover the minimum channel size.
+		{
+			name: "fundmax, local amount < min chan " +
+				"size",
+			initialCoins:       []btcutil.Amount{18_000},
+			selectedCoins:      []btcutil.Amount{18_000},
+			feeRate:            1,
+			chanOpenShouldFail: true,
+			expectedErrStr: "available funds(0.00017877 BTC) " +
+				"below the minimum amount(0.0002 BTC)",
+		},
+		// The local amount exceeds the value of the selected coins.
+		{
+			name: "selected, local amount > " +
+				"selected amount",
+			initialCoins:       []btcutil.Amount{100_000, 50_000},
+			selectedCoins:      []btcutil.Amount{100_000},
+			localAmt:           btcutil.Amount(210_337),
+			chanOpenShouldFail: true,
+			expectedErrStr: "not enough witness outputs to " +
+				"create funding transaction, need 0.00210337 " +
```

Comment:
q: does this error msg account for the reserve it the wallet tho, doesn't look like it ?

Reply:
No it doesn't, in this test case the user chooses too much local_amt for what they have in their wallet. And it's not an anchor but a static remote key channel, so no reserve.